lib/anyway/config.rb

```diff
@@ -44,16 +44,21 @@ def config_name(val = nil)
       #   my_config = Anyway::Config.for(:my_app)
       #   # will load data from config/my_app.yml, secrets.my_app, ENV["MY_APP_*"]
       def for(name)
-        new(name, false).load_from_sources
+        new(config_name: name, do_load: false).load_from_sources
       end
     end
 
     attr_reader :config_name
 
-    def initialize(config_name = nil, do_load = true)
+    def initialize(config_name: nil, do_load: true, config: {})
```

Comment:
Let's use the following signature `initialize(explicit_values = {}, name: nil, load: true)`.

Don't we want to have explicitly provided values non-overridable by other sources (file, env)?

I mean, if I do something like `config = Config.new({a: 1})` I want to make sure that the value of `config.a` doesn't change. Am I right?

Then we have to apply these `explicit_values` after everything is loaded.

Also, we should implement the same behaviour for `#reload` method (and we can re-use it in `#initialize`):

```ruby
def initialize(explicit_values = {}, name: nil, reload: true)
  @config_name = config_name || self.class.config_name
  raise ArgumentError, "Config name is missing" unless @config_name
  reload(explicit_values) if reload
end

def reload(explicit_values = {})
  clear
  load(explicit_values)
  self
end

def load(explicit_values = nil)
  config = load_from_sources((self.class.defaults || {}).deep_dup)
  config.merge!(explicit_values) unless explicit_values.nil?
  config.each do #...
end

Reply:
Hi, this kind of signature is not good for the case, because of it makes you to pass all the arguments if you want ruby to recognize it properly:
```ruby
def test(a = {}, b: 'b', c: 'c')
  puts a.inspect
  puts b
  puts c
end

test()
test({sample: 'string'}) #expect sample be the key ArgumentError: unknown keyword: sample like the next
test(sample: 'string')
test({sample: 'string'}, b: 1, c: 2) #works as expected, but needs to pass other args to make ruby recognize first hash as 'a' param
```

So I suppose to make signature like this:
```ruby
def initialize(explicit_values: {}, name: nil, load: true)
end
```
to allow to pass only needed arguments

Reply:
So in those case i would also like to change the signatures of methods `load` and `reload` to look similar:
```ruby
def load(explicit_values: {})
end
def reload(explicit_values: {})
end
```